server/coordinator/shard_picker.go

```diff
@@ -29,67 +27,68 @@ func NewRandomBalancedShardPicker() ShardPicker {
 }
 
 // PickShards will pick a specified number of shards as expectShardNum.
-func (p *RandomBalancedShardPicker) PickShards(_ context.Context, snapshot metadata.Snapshot, expectShardNum int, enableDuplicateNode bool) ([]storage.ShardNode, error) {
+func (p *RandomBalancedShardPicker) PickShards(_ context.Context, snapshot metadata.Snapshot, expectShardNum int) ([]storage.ShardNode, error) {
 	shardNodes := snapshot.Topology.ClusterView.ShardNodes
 
 	nodeShardsMapping := make(map[string][]storage.ShardNode, 0)
```

Comment:
Add a conditional judgment here to avoid division by 0
```suggestion
	if len(shardNodes) == 0 {
	return storage.ShardNode{}, ErrShardNotEnough
	}
```